sdk/src/v2_api/reader.rs

```diff
@@ -0,0 +1,267 @@
+                return Self::from_manifest_data_and_stream(&manifest_data, &format, &mut file);
+            }
+        }
+        result
+    }
+
+    /// Create a manifest store Reader from a JSON string
+    /// # Arguments
+    /// * `json` - The JSON string
+    /// # Returns
+    /// A reader for the manifest store
+    /// # Note
+    /// This should only be used for testing
+    /// Any referenced resources will not be available
+    pub fn from_json(json: &str) -> Result<Reader> {
+        let manifest_store = serde_json::from_str(json)?;
+        Ok(Reader { manifest_store })
+    }
+
+    /// Create a manifest store Reader from existing c2pa_data and a stream
+    /// You can use this to validate a remote manifest or a sidecar manifest
+    /// # Arguments
+    /// * `c2pa_data` - The c2pa data (a manifest store in JUMBF format)
+    /// * `format` - The format of the stream
+    /// * `stream` - The stream to verify the store against
+    /// # Returns
+    /// A reader for the manifest store
+    /// # Errors
+    /// If the c2pa_data is not valid, or severe errors occur in validation
+    /// validation status should be checked for non severe errors
+    #[async_generic(async_signature(
+        c2pa_data: &[u8],
+        format: &str,
+        mut stream: impl Read + Seek + Send,
+    ))]
+    pub fn from_manifest_data_and_stream(
+        c2pa_data: &[u8],
+        format: &str,
+        mut stream: impl Read + Seek + Send,
+    ) -> Result<Reader> {
+        let mut validation_log = DetailedStatusTracker::new();
+
+        // first we convert the JUMBF into a usable store
+        let store = Store::from_jumbf(c2pa_data, &mut validation_log)?;
+
+        if _sync {
+            Store::verify_store(
+                &store,
+                &mut ClaimAssetData::Stream(&mut stream, format),
+                &mut validation_log,
+            )?;
+        } else {
+            Store::verify_store_async(
+                &store,
+                &mut ClaimAssetData::Stream(&mut stream, format),
+                &mut validation_log,
+            )
+            .await?;
+        }
+
+        Ok(Reader {
+            manifest_store: ManifestStore::from_store(&store, &validation_log),
+        })
+    }
+
+    /// Get the manifest store as a JSON string
+    pub fn json(&self) -> String {
+        self.manifest_store.to_string()
+    }
+
+    /// Get the validation status of the manifest store
+    /// # Returns
+    /// The validation status of the manifest store
+    /// # Example
+    /// ```no_run
+    /// use c2pa::Reader;
+    /// let stream = std::io::Cursor::new(include_bytes!("../../tests/fixtures/CA.jpg"));
+    /// let reader = Reader::from_stream("image/jpeg", stream).unwrap();
+    /// let status = reader.validation_status();
+    /// ```
+    /// # Note
+    /// The validation status should be checked for validation errors`
+    pub fn validation_status(&self) -> Option<&[ValidationStatus]> {
```

Comment:
Should we mention that we only report error codes here? The C2PA spec considers validation status to be made up of both success or error codes.

Related to this, I had to fix a bug in Verify where someone was dragging an asset that had an ingredient with a `validationStatus` showing both error and success codes (which is valid I believe), and since Verify is used to that just being error codes, reported an error since the array wasn't empty. This is all to say people may not realize they will only get failure codes here.

Reply:
Yes it should explain that in the comments. Note that this validation_status is not on an ingredient, but on a manifest_store. Maybe we should have a configuration option to keep the valid status information, but it will tend to bloat the ingredients.